File: data/src/main/java/js/task/data/Repository.java
package js.task.data;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import js.task.data.local.db.model.DataModel;


public interface Repository
{
    Observable<List<DataModel>> getData();

    void updateData(List<DataModel> data);

    void setData(List<DataModel> data);
}

File: data/src/main/java/js/task/data/Repository.kt
package js.task.data

import io.reactivex.Observable
import js.task.data.local.db.model.DataModel


interface Repository
{
    fun getData() : Observable<List<DataModel>>
    fun updateData(data : List<DataModel>)
    fun setData(data : List<DataModel>)
}